### PR TITLE
Update tests to check PRO user and fix existing ones

### PR DESCRIPTION
**Pull request by amurKontur**
## Summary by CodeRabbit

- **Bug Fixes**
	- Enhanced the reliability of population popup data checks by ensuring elements are visible before interaction.

- **Refactor**
	- Improved navigation action methods for clarity and efficiency.
	- Added new method `checkThereIsNoMap()` to verify the absence of a map element.
	- Streamlined the login process by simplifying input handling.
	- Introduced a global setup function to validate essential environment variables.

- **New Features**
	- Introduced automated tests for professional user scenarios, validating critical functionalities within the application.
	- Added support for professional user configurations in the testing framework.
	- Enhanced the authentication process with a new streamlined method for logging in.
	- Added a new testing strategy to verify map accessibility based on project type.
	- Improved toolbar features visibility for different user types and enhanced navigation logic.
	- Updated documentation for improved clarity on running Playwright end-to-end tests.

### CODE REVIEW

File: e2e/login.spec.ts [hunk 1]
```diff
@@ -15,10 +17,13 @@ for (const project of projects) {
       process.env.PASSWORD!,
       50,
     );
+    // TO DO: activate this check once 19103 issue is done
+    // expect(page.url()).toContain('autotests');
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/login.spec.ts:21]: yes please

File: README.md [hunk 2]
```diff
@@ -223,3 +223,31 @@ npx playwright test --ui
 Remove the `--ui` flag to run the tests in headless mode.
 
 > Check the `tests-examples/demo-todo-app.spec.ts` file for an example of how to write tests.
+
+## Running playright tests with locally
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply README.md]: fixed, soon will be pushed

File: README.md [hunk 3]
```diff
@@ -223,3 +223,31 @@
...
+
+1. Setup the .env.playwright.local file
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply README.md]: fixed locally will be pushed soon

File: README.md [hunk 4]
```diff
@@ -223,3 +223,31 @@
...
+EMAIL_PRO=<pro-test-email>
+PASSWORD_PRO=<pro-test-password>
+ENVIRONMENT=local-<env>
+APP_NAME=all
+ADMIN_KEYCLOAK=<keycloak-login>
+ADMIN_KEYCLOAK_PASSWORD=<keycloak-password>
+SLACK_BOT_USER_OAUTH_TOKEN=token
+```
+
+Where <env> is the environment your local app is configured, for example `local-dev`.
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply README.md]: it is not html

File: e2e/auth.setup.ts [hunk 5]
```diff
@@ -1,36 +1,23 @@
 import { test as setup } from '@playwright/test';
-import { expect } from '@playwright/test';
 import { getProjects } from './page-objects/helperBase.ts';
 import { PageManager } from './page-objects/pageManager.ts';
 
-setup('Authentication', async ({ page, context }) => {
+setup('Authentication', async ({ context }) => {
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/auth.setup.ts:5]: we have global-setup assertion for that

File: e2e/authAsPro.setup.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { test as setup } from '@playwright/test';
+import { getProjects } from './page-objects/helperBase.ts';
+import { PageManager } from './page-objects/pageManager.ts';
+
+setup('Authentication with PRO', async ({ context }) => {
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/authAsPro.setup.ts:5]: we have global-setup assertion for that

File: e2e/global-setup.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    'SLACK_BOT_USER_OAUTH_TOKEN',
+  ];
+
+  const sendErrorIfUndefined = function (param: string) {
+    if (typeof process.env[param] === 'undefined')
+      throw new Error(`${param} is not defined at .env file`);
+  };
+
+  params.forEach((param) => sendErrorIfUndefined(param));
+}
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/global-setup.ts:20]: you don't use variables that were created during destructuring. And I also don't see the point in them. Your comment makes no sense

File: e2e/locationWithPro.spec.ts [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+for (const project of projects) {
+  // Setting geolocation permission and geolocation of user's browser
+  test.use({
+    permissions: ['geolocation'],
+    geolocation: { longitude: -74.0324, latitude: 40.714 },
+  });
+
+  // Prod has other dimensions of screenshots so test fails if using wrong screenshot. Screenshots are created by test name, browser and OS by playwright
+
+  if (project.env != 'prod') {
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/locationWithPro.spec.ts]: fixed locally will be pushed soon

File: e2e/page-objects/loginPage.ts [hunk 9]
```diff
@@ -1,39 +1,56 @@
...
-    const emailInput = this.page.getByRole('textbox').first();
-    await emailInput.pressSequentially(email, { delay: typingSpeedMs });
+    await this.checkLoginAndSignupPresence(operablePage);
+    // Getting email field and type in
+    const emailInput = operablePage.getByRole('textbox').first();
+    await emailInput.fill(email);
+
+    // Getting password field and type in
+    const passwordInput = operablePage.locator('input[type="password"]');
+    await passwordInput.fill(password);
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/page-objects/loginPage.ts:34]: getByRole is a special playwright method

File: e2e/page-objects/navigationMenu.ts [hunk 10]
```diff
@@ -1,40 +1,71 @@
...
-    await this.page.locator('div', { hasText: 'Collapse' }).getByText('Login').click();
+  async goToLoginPage(operablePage: Page = this.page) {
+    // TO DO: replace locator here once 19141 task is done, refactor this logics
+    // When no need will be present to get locator every time
+    await operablePage.locator('[value="profile"]').first().waitFor({ state: 'visible' });
+    await operablePage.locator('[value="profile"]').getByText('Login').hover();
+    await operablePage
+      .locator('[value="profile"]')
+      .getByText('Login')
+      .click({ delay: 330, force: true });
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/page-objects/navigationMenu.ts:40]: it is then used for the next locator so finally it is unique

File: e2e/page-objects/pageManager.ts [hunk 11]
```diff
@@ -1,3 +1,4 @@
+import { expect } from '@playwright/test';
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/page-objects/pageManager.ts:1]: are you nuts?

File: e2e/page-objects/pageManager.ts [hunk 12]
```diff
@@ -62,4 +64,22 @@
...
+   */
+
+  async auth(project: Project, email: string, password: string, newPage: Page) {
+    await this.atBrowser.openProject(project, { operablePage: newPage });
+    await this.fromNavigationMenu.goToLoginPage(newPage);
+    await this.atLoginPage.typeLoginPasswordAndLogin(email, password, {
+      shouldSuccess: true,
+      project,
+      operablePage: newPage,
+    });
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/page-objects/pageManager.ts]: changed the name to operablePage

File: README.md [hunk 13]
```diff
@@ -195,31 +195,66 @@
...
+PASSWORD_PRO=<pro-test-password>
+ENVIRONMENT=local-<env>
+APP_NAME=all
+ADMIN_KEYCLOAK=<keycloak-login>
+ADMIN_KEYCLOAK_PASSWORD=<keycloak-password>
+SLACK_BOT_USER_OAUTH_TOKEN=token
 ```
 
-Set your projects environment at projects-config.json file
+Where <env> is the environment your local app is configured, for example `local-dev`.
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply README.md]: it is not html

File: README.md [hunk 14]
```diff
@@ -195,31 +195,66 @@ You can your own for override default config, it should have name `./configs/con
...
+EMAIL_PRO=<pro-test-email>
+PASSWORD_PRO=<pro-test-password>
+ENVIRONMENT=<env>
+APP_NAME=all
+ADMIN_KEYCLOAK=<keycloak-login>
+ADMIN_KEYCLOAK_PASSWORD=<keycloak-password>
+SLACK_BOT_USER_OAUTH_TOKEN=token
+```
+
+Where <env> is the environment you want to test, for example `test`.
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply README.md:216]: it is not html

**Review by dqunbp — APPROVED** [review]

**Pull request merged**